client/src/pages/private/SiteTable.js

```diff
@@ -354,16 +192,47 @@ export default ({ sites, viewOnly }) => {
 
   const openNewPhaseModal = () => {
     closeActionMenu();
-    // TODO HCAP-1277 setActiveModalForm('new-phase');
+    setActiveModalForm('new-phase');
+  };
+
+  const handleFormSubmit = async (submitFunction, formData) => {
+    const response = await submitFunction(formData);
+    if (response.ok) {
+      closeDialog();
+      await fetchSites();
+    }
+    if (response.toast) {
+      openToast(response.toast);
+    }
   };
 
   return (
     <>
-      <SiteFormsDialog
-        activeForm={activeModalForm}
-        onDialogSubmit={fetchSites}
-        onDialogClose={closeDialog}
-      />
+      {activeModalForm === 'new-site' && (
```

Comment:
I agree this still looks a bit chunky, I don't think the conditionals are necessary, and what do you think of moving the dialogs into the same component as the form `NewSiteDialog`?

Will these forms ever be rendered outside of a dialog?

Reply:
The one thing that I don't like about it is that it is inconsistent with the rest of the codebase- we have a whole directory full of Form components that probably all go into a Dialog in every use case.